cirq-core/cirq/ops/dense_pauli_string.py

```diff
@@ -180,19 +174,20 @@ def _resolve_parameters_(self: TCls, resolver: 'cirq.ParamResolver', recursive:
     def __pos__(self):
         return self
 
-    def __pow__(self, power):
+    def __pow__(self: TCls, power: Union[float, int]) -> TCls:
         if isinstance(power, int):
             i_group = [1, +1j, -1, -1j]
             if self.coefficient in i_group:
                 coef = i_group[i_group.index(self.coefficient) * power % 4]
             else:
                 coef = self.coefficient ** power
             if power % 2 == 0:
-                return coef * DensePauliString.eye(len(self))
-            return DensePauliString(coefficient=coef, pauli_mask=self.pauli_mask)
+                return coef * self.eye(len(self))
```

Comment:
This should be `type(self).eye` since `eye` is a class method. We should also have a test to verify this works as expected. 
```suggestion
                return coef * type(self).eye(len(self))
```

Reply:
- `self.eye` and `type(self).eye` are semantically equivalent here. Is your suggestion referring to some convention or best practice?
- I'll add the test

Reply:
I changed to `type(self)` as suggested.